src/ResponseTagger/Delegator/XLocationIdBCTagger.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license   For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\PlatformHttpCacheBundle\ResponseTagger\Delegator;
+
+use EzSystems\PlatformHttpCacheBundle\ResponseConfigurator\ResponseCacheConfigurator;
+use EzSystems\PlatformHttpCacheBundle\ResponseTagger\ResponseTagger;
+use Symfony\Component\HttpFoundation\Response;
+
+class XLocationIdBCTagger implements ResponseTagger
+{
+    /**
+     * @var ResponseTagger
+     */
+    private $locationTagger;
+
+    const LOCATION_ID_HEADER = 'X-Location-Id';
+
+    /**
+     * XLocationIdTagger constructor.
+     *
+     * @param ResponseTagger $locationTagger
+     */
+    public function __construct(ResponseTagger $locationTagger) {
+        $this->locationTagger    = $locationTagger;
+    }
+
+    public function tag(ResponseCacheConfigurator $configurator, Response $response, $view)
+    {
+        if ($view !== null || !$response->headers->has(static::LOCATION_ID_HEADER)) {
+            return null;
+        }
+
+        $locationIds = explode("|", $response->headers->get(static::LOCATION_ID_HEADER));
+        foreach ($locationIds as $locationId) {
+            $location = $this->repository->getLocationService()->loadLocation($locationId);
```

Comment:
After internal discussions, 

Did you consider going for a simpler tagging, and only generate `location-<locationId>` ? It would avoid having to load the location.

On the long term, this will have to be fixed in the proper layers anyway. For instance, we have two pull-requests about REST & HTTP caching, both in regards to REST includes: https://github.com/ezsystems/ezpublish-kernel/pull/1741 & https://github.com/ezsystems/ezpublish-kernel/pull/1789 (WIP). They've been here for a while, but do touch the same areas.

As an alternative, an option would be to override the `CachedValue` value object visitor from the REST server. This is where we generate the `X-Location-ID` header. As for available data, we have both the location ID _and_ the the cached value object, meaning that we can use Response Taggers.